engine/model.py

```diff
@@ -309,3 +313,8 @@ def tile_width(self) -> int:
     def tile_height(self) -> int:
         """Gets the tile height of the map in pixels."""
         return self.tilemaps[self.active_region].tile_height
+
+    @property
+    def game_time(self) -> int:
```

Comment:
Unfortunately this is not quite accurate: the docstring says it's in seconds but the function is returning milliseconds.

Let's keep things in milliseconds for performance reasons, so three changes requested here:
1) Update the docstring.
2) Rename the property to `game_time_millis`
3) Remove the `int()` call. It's redundant.

Reply:
Sorry if I'm missing something, but `delta_time` is always around 0.0168, which is 1/60 of a second, so isn't it in seconds?

Reply:
Got it, I thought it was in millis since it was an int. In that case we should update `on_update` to be a float.

As a side note: it won't _always_ be 0.0168, it will be that value whenever the system is not under load. If we put too many things in the game and it takes more than 16.8ms to process the scene, then `delta_time` will increase. The purpose of using a variable like `delta_time` instead of just hard-coding 1 / 60 for everything is so that even when the frame rate drops momentarily, the game will still appear to progress at the same rate.

An example: humans won't notice too too much if the frame rate drops down to 30 fps, but since that's twice as slow as the expected frame rate if you assume `on_update` is called 60 times per second it will slow down everything by half. Animation speeds, game time, etc. That would be _very_ noticeable to a human if everything started to move in slow motion.

Reply:
Alright, makes sense Why a float though? Do we need the time down to the millisecond?

Reply:
I don't know if we'd need time down right to the millisecond, but we'll absolutely need sub-second resolution. This game time piece is the way the engine tells the game logic about time, so _any_ game logic that wants to use time will use this. One that comes to mind immediately that would use sub-second resolution is attack cooldowns: things like weapons or spells that want to fire more frequently than once per second (which is quite slow).

Reply:
Haha that's what everyone means when they say a senior dev has the whole picture in their head. Thanks for the clarification, and I've made the necessary changes already.